fe/fe-core/src/main/java/com/starrocks/catalog/HiveTable.java

```diff
@@ -194,34 +210,51 @@ public Map<String, HiveColumnStats> getTableLevelColumnStats(List<String> column
         return HiveMetaStoreTableUtils.getTableLevelColumnStats(hmsTableInfo, columnNames);
     }
 
+    public void refreshExternalTableSchema(org.apache.hadoop.hive.metastore.api.Table table) throws DdlException {
+        Map<String, FieldSchema> updatedTableSchemas = HiveMetaStoreTableUtils.getAllHiveColumns(table);
+        refreshSchemaCache(table.getSd().getCols(), updatedTableSchemas);
+        modifyTableSchema(srFullDbName, getName());
```

Comment:
You should make sure this table hav't been dropped before write editlog. Or it will throw npe when replaying the OP_MODIFY_HIVE_TABLE_COLUMN. So you should operate this in db read lock.